toolkit/tools/imagecustomizerapi/artifacts.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import "fmt"
+
+type Artifacts struct {
+	Items []Item `yaml:"items" json:"items,omitempty"`
+	Path  string `yaml:"path" json:"path,omitempty"`
+}
+
+func (a Artifacts) IsValid() error {
+	if (len(a.Items) == 0 && a.Path != "") || (len(a.Items) > 0 && a.Path == "") {
```

Comment:
`Artifacts` is a pointer in the `Output` struct. So, you could just enforce that both `Items` and `Path` are not empty.